cypress/integration/ui_spec.js

```diff
@@ -0,0 +1,685 @@
+					win.parleySettings.authHeader = newAuthHeader;
+				});
+
+				cy.wait("@createDevice");
+			});
+		});
+		describe("userAdditionalInformation", () => {
+			it("should re-register the device when it changes", () => {
+				const parleyConfig = {
+					roomNumber: "0W4qcE5aXoKq9OzvHxj2",
+					userAdditionalInformation: {"some-key": "some-value"},
+				};
+				const testMessage = `test message before switching userAdditionalInformation ${Date.now()}`;
+
+				cy.visit("/", {
+					onBeforeLoad: (win) => {
+						// eslint-disable-next-line no-param-reassign
+						win.parleySettings = parleyConfig;
+					},
+				});
+
+				cy.get("[id=app]").as("app");
+
+				clickOnLauncher();
+				sendMessage(testMessage);
+				findMessage(testMessage); // Wait until the server received the new message
+
+				// Test if it changes during runtime
+				const newUserAdditionalInformation = {
+					"some-key": "some-value",
+					"some-layer": {"some-key-in-layer": "some-value-in-layer"},
+				};
+
+				cy.intercept("POST", "*/**/devices", (req) => {
+					expect(JSON.parse(req.body))
+						.to.have.deep.property("userAdditionalInformation", newUserAdditionalInformation);
+				}).as("createDevice");
+
+				cy.window().then((win) => {
+					// eslint-disable-next-line no-param-reassign
+					win.parleySettings.userAdditionalInformation["some-layer"] = newUserAdditionalInformation["some-layer"];
+				});
+
+				cy.wait("@createDevice");
+			});
+		});
+		describe("weekdays", () => {
+			describe("format [day, start, end]", () => {
+				it("should show we are offline/online outside/inside working hours", () => {
+					const parleyConfig = {
+						weekdays: [ // closed every day
+							["Monday"],
+							["Tuesday"],
+							["Wednesday"],
+							["Thursday"],
+							["Friday"],
+							["Saturday"],
+							["Sunday"],
+						],
+						interface: {hideChatAfterBusinessHours: true},
+					};
+
+					cy.visit("/", {
+						onBeforeLoad: (win) => {
+							// eslint-disable-next-line no-param-reassign
+							win.parleySettings = parleyConfig;
+						},
+					});
+
+					cy.get("[id=app]").as("app");
+
+					// Launcher is not rendered because we are offline
+					// and outside working hours
+					cy.get("@app")
+						.get("[class^=launcher__]")
+						.should("not.exist");
+
+					// Test if it changes during runtime
+					const newWeekdays = [
+						[
+							"Monday", 8.00, 23.30,
+						],
+						[
+							"Tuesday", 8.00, 23.30,
+						],
+						[
+							"Wednesday", 8.00, 23.30,
+						],
+						[
+							"Thursday", 8.00, 23.30,
+						],
+						[
+							"Friday", 8.00, 23.30,
+						],
+						[
+							"Saturday", 8.00, 23.30,
+						],
+						[
+							"Sunday", 8.00, 23.30,
+						],
+					];
+
+					cy.window().then((win) => {
+						// eslint-disable-next-line no-param-reassign
+						win.parleySettings.weekdays = newWeekdays;
+					});
+
+					// Launcher should appear again because we
+					// are inside working hours
```

Comment:
Hoe zorg je ervoor dat je ook tussen 23:00 en 08:00 bij het uitvoeren van deze test inside working hours zit?

Reply:
Heb er `0.00, 23.59` van gemaakt zodat ie 24 uur open is 2f3ZfCeFQb4Lj3qj7dCjqyvi8D8VrgVM4mujzhscXvLA8sCNsfqMGuiXMteRrrN3